emukit/quadrature/methods/warpings.py

```diff
@@ -1,30 +1,34 @@
+import abc
 import numpy as np
 from typing import Optional
 
 
-class Warping:
+class Warping(abc.ABC):
     """The warping for warped Bayesian quadrature."""
 
+    @abc.abstractmethod
     def transform(self, Y: np.ndarray) -> np.ndarray:
         """Transform from base-GP to integrand.
 
         :param Y: Function values of latent function, shape (n_points, 1).
         :return: Transformed values, shape (n_points, 1).
         """
-        raise NotImplemented
+        pass
 
+    @abc.abstractmethod
     def inverse_transform(self, Y: np.ndarray) -> np.ndarray:
         """Transform from integrand to base-GP.
 
         :param Y: Function values of integrand, shape (n_points, 1).
         :return: Transformed values, shape (n_points, 1).
         """
-        raise NotImplemented
-
-    def update_parameters(self, **kwargs) -> None:
-        """Update the warping parameters."""
         pass
 
+    def update_parameters(self, **new_parameters) -> None:
+        """Update the warping parameters. The keyword arguments ``new_parameters`` contain the parameter names as
+        keys with the new values. An empty dictionary will not update any parameters."""
+        self.__dict__.update(new_parameters)
```

Comment:
Seems kinda scary to mess with this. Is there any reason why we can't just create `_warping_parameters = {}` and update that?

Reply:
I'll replace it with 
```
for parameter, new_value in new_parameters.items():
    setattr(self, parameter, new_value)
```

That is more explicit. In what way is it scary?

Reply:
i was always assuming this is a read-only thingy. it probably partially generated on the fly in Python internals

Reply:
There is a test for it in the test file for warpings so I was confident it at least did the job functionality wise, but I agree it looked a bit suspicious. Couldn't find a good reason though. `setattr` seems better though.

Reply:
Yes, setattr looks more reasonable